hpx/parallel/execution_policy.hpp

```diff
@@ -162,24 +163,23 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
         ///                     execution of the parallel algorithm the
         ///                     returned execution policy is used with.
         ///
-        /// \note Requires: is_executor_parameters<Parameters>::value is true
+        /// \note Requires: all parameters are executor_parameters,
+        ///                 different parameter types can't be duplicated
         ///
         /// \returns The new sequential_task_execution_policy
         ///
-        template <typename Parameters>
+        template <typename... Parameters, typename ParametersType =
+            typename hpx::parallel::executor_parameters_join<Parameters...>::type>
```

Comment:
What's the rationale for making the `ParametersType` a template argument? Would it be sufficient to directly use the `hpx::parallel::executor_parameters_join<Parameters...>` in the body of this function?